src/rt-app_parse_config.c

```diff
@@ -811,7 +811,7 @@ parse_thread_data(char *name, struct json_object *obj, int index,
 		data->phases = malloc(sizeof(phase_data_t) * data->nphases);
 		parse_thread_phase_data(obj,  &data->phases[0], opts, (long)data);
 		/* Get loop number */
-		data->loop = -1;
+		data->loop = get_int_value_from(obj, "loop", TRUE, -1);
 	}
```

Comment:
the problem is : 
with json like below
"thread0" : {
"run" : 10000,
}
phase->loop will be 1 and thread->loop must be -1 to loop indefinitely

with json like below
"thread0" : {
"loop": X
"run" : 10000,
}
phase->loop will be X and thread->loop must be 1 to loop only X times.

AFAICT, there is no problem as soon as we have a phase object, isn't it ?

So may be do:
@@ -811,7 +811,11 @@ parse_thread_data(char *name, struct json_object *obj, int index,
 		data->phases = malloc(sizeof(phase_data_t) * data->nphases);
 		parse_thread_phase_data(obj,  &data->phases[0], opts, (long)data);
 		/* Get loop number */
-		data->loop = get_int_value_from(obj, "loop", TRUE, -1);
+		if(get_in_object(obj, "loop", TRUE))
+			data->loop = 1;
+		else
+			data->loop = -1;
+
 	}
 
 }